x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -508,9 +517,18 @@ public boolean isActive() {
     public boolean checkFeature(Feature feature) {
         boolean allowed = isAllowed(feature);
         LongAccumulator maxEpochAccumulator = lastUsed.get(feature);
+        long now = System.currentTimeMillis();
         if (maxEpochAccumulator != null) {
             maxEpochAccumulator.accumulate(epochMillisProvider.getAsLong());
         }
+
+        if(feature.minimumOperationMode.compareTo(OperationMode.BASIC) > 0
+            && now > status.licenseExpiryDate - GRACE_PERIOD_DURATION.getMillis()) {
```

Comment:
I think we'd need to access this by calling `checkAgainstStatus` instead of accessing the licenseExpiryDate  field directly